docs/DeveloperGuide.md

```diff
@@ -11,10 +11,25 @@ Given below is a quick overview of the main components of Mod Happy and how they
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Tag Feature
+
+The tag command accepts a string from the user and adds it into `ArrayList<String> tags` of a `Task`.  
+
+Here is an example on adding a tag to a general task:  
+1) User inputs `tag add 2 "testTag"`. 
+2) `TagParser` will initialise `TagCommand` with add as `tagOperation` 2 as `taskIndex` and testTag as `tagDescription`, while `taskModule` is null.
+3) `TagCommand` then gets the relevant `Module`. If `taskModule` is null, `getGeneralTasks()` is called. Else, `getModule(taskModule)` is called instead.
+4) Next, `TagCommand` checks the `tagOperation`. If add, `addTag(targetModule)` is called. Else if del, `removeTag(targetModule)` is called. Else, it throws `ParseException`.
+
+Below is the sequence diagram of how the tag feature works:
+
+![Tag](https://user-images.githubusercontent.com/70083643/159489546-574538e0-ee99-40c0-b373-1acc1542105d.png)
+![GetModule](https://user-images.githubusercontent.com/70083643/159489557-d82be092-7570-459c-b933-5d1497490b98.png)
+![ExecuteTagOp](https://user-images.githubusercontent.com/70083643/159489602-2a88ed8f-3227-4f06-bf4a-d1f743d60856.png)
```

Comment:
Should I change these to the link that the rest of you are using? Or is this better?

Reply:
Yep, I suggest so. The reason is that later it will automatically update when you change the puml file